pkg/util/kubernetes/autoscalers/processor.go

```diff
@@ -131,14 +143,53 @@ func (p *Processor) ProcessWPAs(wpa *v1alpha1.WatermarkPodAutoscaler) map[string
 	return externalMetrics
 }
 
+func makeChunks(batch []string) (chunks [][]string) {
+	for i := 0; i < len(batch); i += chunkSize {
+		if i+chunkSize > len(batch) {
+			chunks = append(chunks, batch[i:])
+			break
+		}
+		chunks = append(chunks, batch[i:i+chunkSize])
+	}
+	return chunks
+}
+
 // validateExternalMetric queries Datadog to validate the availability and value of one or more external metrics
 func (p *Processor) validateExternalMetric(emList map[string]custommetrics.ExternalMetricValue) (processed map[string]Point, err error) {
-	var batch []string
+	batch := []string{}
 	for _, e := range emList {
 		q := getKey(e.MetricName, e.Labels)
 		batch = append(batch, q)
 	}
-	return p.queryDatadogExternal(batch)
+	chunks := makeChunks(batch)
+	log.Tracef("List of batches %v", chunks)
+
+	// we have a number of chunks with `chunkSize` metrics.
+	responses := make(chan queryResponse, len(batch))
+	processed = make(map[string]Point)
+	errors := []error{}
```

Comment:
a minor nit:
```suggestion
	var errors []error
```
also it can be declare just before line 183 https://github.com/DataDog/datadog-agent/pull/4617/files#diff-31878728b8b72cdd470c0d6bb81f3d56R183

Reply:
good  point